v2/protocol/http/protocol_lifecycle.go

```diff
@@ -38,8 +38,9 @@ func (p *Protocol) OpenInbound(ctx context.Context) error {
 	}
 
 	p.server = &http.Server{
-		Addr:    listener.Addr().String(),
-		Handler: attachMiddleware(p.Handler, p.middleware),
+		Addr:              listener.Addr().String(),
+		Handler:           attachMiddleware(p.Handler, p.middleware),
+		ReadHeaderTimeout: DefaultReadTimeout,
```

Comment:
Question: How about we fix it for both read/write? Or at least set `ReadTimeout` instead of just for headers to [protect the read path](https://ieftimov.com/posts/make-resilient-golang-net-http-servers-using-timeouts-deadlines-context-cancellation/)?

Reply:
Good point - I'll do both (read and write) and do the entire thing, not just headers